Added URL Compaction algorithm.

gkellogg opened the pull request:
This partially completes #365.

gkellogg committed:
Added URL Compaction algorithm.
This partially completes #365.

6a6d74 committed:
added `id` for url-compaction

6a6d74 committed:
Revert "added `id` for url-compaction"

This reverts commit 9c621b78ffe6c73d128bc7b96a7c0908ea452193.

6a6d74 committed:
added references to url-compaction

URL Compaction _only_ applied to propertyUrl, Common Properties and
Notes … we agreed in ISSUE #365 _not_ to compact aboutUrls or valueUrls

6a6d74 committed:
valueUrls don't get compacted

as agreed in ISSUE #365, valueUrls don’t get compacted - so I’ve
“uncompacted” `schema:` back to `http://schema.org/`

6a6d74 commented:
fixes #365

6a6d74 commented:
@gkellogg - can you check that what I've done matches your expectations? In particular, note that I have only applied compaction to `notes`, `common properties` and `propertyUrl`s ... _NOT_ `aboutUrl`s or `valueUrl`s.

gkellogg committed:
Clarify that values of `@type` used within common properties and notes are compacted as are propertyUrls.

gkellogg committed:
Compact V<sub>URL</sub> if N is `@type`.
This also reverts 788ea4be08f5cd6564fb3cd563c7d9e34aae7197 so that any value of `@type` in JSON is subject to URL compaction.

gkellogg commented:
@6a6d74 I think the last two commits are what we want, so that all values of `@type` will be compacted, whether in common properties/notes or elsewhere.
